op-challenger/challenger/indexer.go

```diff
@@ -0,0 +1,59 @@
+package challenger
+
+import (
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// indexer runs the indexing service for the challenger.
+// This function spawns a goroutine to run the indexer.
+func (c *Challenger) indexer() {
```

Comment:
is this only going forwards?

Reply:
It's dependent upon having a traversal client to iterate over receipts in #5756, so I just "moosed" them as comments.

Open to changing this up ofc, just wanted to mock it out